### ISSUE TITLE
Lexer for ResourceBundle

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#745]
(Original issue 1038 created by None on 2014-10-06T05:25:31.167099+00:00)

This Lexer shoud parse [ICU ResourceBundles](http://userguide.icu-project.org/locale/resources):

```
#!python

class ResourceLexer(RegexLexer):
   name = 'ResourceBundle'
   aliases = ['resource', 'resourcebundle', 'rb']
   filenames = ['*.txt']

   flags = re.MULTILINE | re.IGNORECASE
   tokens = {
      'root': [
         (r'//.*?$', Comment),
         (r'"', String, 'string'),
         (r'-?\d+', Number.Integer),
         (r'[,{}]', Operator),
         (r'([^\s{:]+)(\s*)((?::table|:array|:string|:bin|:import|:intvector|:int|:alias|)?)', bygroups(Name, Text, Keyword)),
         (r'\s+', Text)
      ],
      'string': [
         (r'(\\\\|\\"|\\{|[^"{])+', String),
         (r'{', String.Escape, 'msgname'),
         (r'"', String, '#pop')
      ],
      'msgname': [
         (r'([^{},]+)(\s*)', bygroups(Name, String.Escape), ('#pop', 'message'))
      ],
      'message': [
         (r'{', String.Escape, 'msgname'),
         (r'}', String.Escape, '#pop'),
         (r'(,)(\s*)([a-zA-Z]+)(\s*})', bygroups(Operator, String.Escape, Keyword, String.Escape), '#pop'),
         (r'(,)(\s*)([a-zA-Z]+)(\s*)(,)(\s*)(offset)(\s*)(:)(\s*)(-?\d+)(\s*)', bygroups(Operator, String.Escape, Keyword, String.Escape, Operator, String.Escape, Operator.Word, String.Escape, Operator, String.Escape, Number.Integer, String.Escape), 'choice'),
         (r'(,)(\s*)([a-zA-Z]+)(\s*)(,)(\s*)', bygroups(Operator, String.Escape, Keyword, String.Escape, Operator, String.Escape), 'choice'),
         (r'\s+', String.Escape)
      ],
      'choice': [
         (r'(=|<|>|<=|>=|!=)(-?\d+)(\s*{)', bygroups(Operator, Number.Integer, String.Escape), 'message'),
         (r'([a-zA-Z]+)(\s*{)', bygroups(Keyword.Type, String.Escape), 'str'),
         (r'}', String.Escape, ('#pop', '#pop')),
         (r'\s+', String.Escape)
      ],
      'str': [
         (r'}', String.Escape, '#pop'),
         (r'{', String.Escape, 'msgname'),
         (r'[^{}]+', String)
      ]
   }
```

**Comment by Anteru** [comment]
(Original issue 1038 last updated on 2014-10-15T05:58:51.454397+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-10-14T18:33:17.304131+00:00)

Do you have an example file we can include?

**Comment by Anteru** [comment]
(Original comment by None on 2014-10-14T18:46:41.619648+00:00)

There are lots of examples given on the ICU site. A long one would be [calendar.txt](http://source.icu-project.org/repos/icu/icu/trunk/source/test/testdata/calendar.txt), while a shorter one (taken from [here](http://userguide.icu-project.org/locale/resources)) is

```
#!resourcebundle

root:table {
  usage:string { "Usage: genrb [Options] files" }
  version:int { 122 }
  errorcodes:array {
    :string { "Invalid argument" }
    :string { "File not found" }
  }
}
```
And I've just noticed that in order to support unnamed, typed array entries, the following line should be inserted before the last one in root:

```
#!python

         (r':table|:array|:string|:bin|:import|:intvector|:int|:alias', Keyword),
```

**Comment by Anteru** [comment]
(Original comment by tshatch on 2014-10-15T05:58:51.450189+00:00)

Add ResourceBundle lexer.

Fixes #1038

→ <<cset cafcc2615cc5>>
